__tests__/app.test.js

```diff
@@ -158,4 +158,107 @@ describe('app', () => {
                 });
         });
     })
+    describe("POST /api/articles/:article_id/comments", () => {
+        test('comment is posted and the response is the comment 201', () => {
+            const userComment = {
+                username: "butter_bridge",
+                body: "wow comment"
+            }
+            return request(app)
+                .post('/api/articles/1/comments')
+                .send(userComment)
+                .expect(201)
+                .then(({body}) => {
+                    const {comment} = body;
+                    expect(comment).toMatchObject(
+                        expect.objectContaining({
+                            article_id: 1,
+                            author: "butter_bridge",
+                            body: "wow comment",
+                            comment_id: 19,
+                            created_at: expect.any(String),
+                            votes: 0,
+                        }))
+                });
+        });
+        test("bad request code 400 not a valid user", () => {
+            const comment = {
+                username: "mike",
+                body: "comment",
+            };
+            return request(app)
+                .post("/api/articles/1/comments")
+                .send(comment)
+                .expect(400)
+                .then(({body}) => {
+                    const {msg} = body;
+                    expect(msg).toBe("bad request");
+                });
+        });
+
+        test("bad request 400 invalid articleID", () => {
+            const comment = {
+                username: "mike",
+                body: "comment"
+            };
+            return request(app)
+                .post("/api/articles/9999999/comments")
+                .send(comment)
+                .expect(400)
+                .then(({body}) => {
+                    const {msg} = body;
+                    expect(msg).toBe("bad request");
+                });
```

Comment:
This one would be a 404 for valid but non existent ID. Think you just need to amend the username to a valid one too!